common/primitives/src/messages.rs

```diff
@@ -14,24 +14,31 @@ use utils::*;
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
 pub struct MessageResponse<BlockNumber> {
-	#[cfg_attr(feature = "std", serde(with = "as_hex"))]
-	/// Serialized data in a the schemas.
-	pub payload: Vec<u8>,
 	/// Message source account id of the Provider. This may be the same id as contained in `msa_id`,
 	/// indicating that the original source MSA is acting as its own provider. An id differing from that
 	/// of `msa_id` indicates that `provider_msa_id` was delegated by `msa_id` to send this message on
 	/// its behalf.
 	pub provider_msa_id: MessageSourceId,
-	///  Message source account id (the original source).
-	pub msa_id: MessageSourceId,
 	/// Index in block to get total order.
 	pub index: u16,
 	/// Block-number for which the message was stored.
 	pub block_number: BlockNumber,
+	///  Message source account id (the original source).
+	#[cfg_attr(feature = "std", serde(skip_serializing_if = "Option::is_none", default))]
+	pub msa_id: Option<MessageSourceId>,
+	/// Serialized data in a the schemas.
+	#[cfg_attr(
+		feature = "std",
+		serde(with = "as_hex_option", skip_serializing_if = "Option::is_none", default)
+	)]
+	pub payload: Option<Vec<u8>>,
+	/// The content address for an IPFS payload
+	#[cfg_attr(feature = "std", serde(skip_serializing_if = "Option::is_none", default))]
+	pub cid: Option<Vec<u8>>,
```

Comment:
The Issue shows the CID as the payload when it's an off-chain, IPFS message.  What motivated the design change?

Reply:
I'm confused. The issue does show responding with the `cid` value if it is an IPFS message. Which is what this is right?

Reply:
No actually, I was confused by the way it was written but after a night's sleep I see how I was mistaken.